x/ccv/consumer/keeper/relay.go

```diff
@@ -104,43 +94,26 @@ func (k Keeper) SendVSCMaturedPackets(ctx sdk.Context) error {
 	for maturityIterator.Valid() {
 		vscId := types.IdFromPacketMaturityTimeKey(maturityIterator.Key())
 		if currentTime >= binary.BigEndian.Uint64(maturityIterator.Value()) {
-			// send VSCMatured packet
-			// - construct validator set change packet data
-			packetData := ccv.NewVSCMaturedPacketData(vscId)
-			// - send packet over IBC
-			err := utils.SendIBCPacket(
-				ctx,
-				k.scopedKeeper,
-				k.channelKeeper,
-				channelID,          // source channel id
-				ccv.ConsumerPortID, // source port id
-				packetData.GetBytes(),
-				k.GetCCVTimeoutPeriod(ctx),
-			)
-			if err != nil {
-				return err
-			}
+			// construct validator set change packet data
+			vscPacket := ccv.NewVSCMaturedPacketData(vscId)
+
+			// append VSCMatured packet to pending packets
+			// sending packets is attempted each EndBlock
+			// unsent packets remain in the queue until sent
+			k.AppendPendingPacket(ctx, types.ConsumerPacket{
+				Type: types.VscMaturedPacket,
+				Data: vscPacket.GetBytes(),
+			})
 			k.DeletePacketMaturityTime(ctx, vscId)
-			ctx.EventManager().EmitEvent(
```

Comment:
Why is this event deleted?

Reply:
There's no sending going on. The packets are only queued for sending.

It would make more sense to add logging here and emit an event when the packet is sent.
Since we store all packets in the same queue, we'd need to unmarshal the data, check that it contains a maturedVSC packet and use it to build the event. The event seemed to have low value so it was dropped.

I'll reintroduce it in `SendPackets`

Reply:
Yes, that's the event that we care about. The actual packet being sent it's an IBC event anyways.